RecoTauTag/RecoTau/plugins/DeepTauId.cc

```diff
@@ -630,9 +630,21 @@ namespace {
     auto getTauDxy(const pat::PackedCandidate& cand, float default_value) { return cand.dxy(); }
     auto getPvAssocationQuality(const reco::PFCandidate& cand) { return 0.7013f; }
     auto getPvAssocationQuality(const pat::PackedCandidate& cand) { return cand.pvAssociationQuality(); }
-    auto getPuppiWeight(const reco::PFCandidate& cand) { return 0.9907f; }
+    auto getPuppiWeight(const reco::PFCandidate& cand) { 
+      if (std::abs(cand.pdgId()) == 11) return 0.9787f;
+      else if (std::abs(cand.pdgId()) == 13) return .9907f;
```

Comment:
Please add 0 before decimal point (here and other new lines where it is missing)
Please also confirm that you have run
```
scram b code-check
scram b code-format
```
as I am not sure if current formatting agrees with CMS rules.
BTW, literally, `else` statements are not mandatory here because of `return` executed when condition is fulfilled. Maybe `case` is nicer construction here? With `case`  `std::abs(cand.pdgId())` will be evaluated just once which is also a plus.

Reply:
Done

Reply:
Thanks.
Could you please put zero before decimal point consistently for all default values here and in the `getPuppiWeightNoLep` method? It can be done together with other changes requested by Konstantin.

Reply:
Cosmetics: maybe it makes also sense to order pdgIds in the same way in both methods, e.g. (11,13) 22, 211, 130.
BTW, is it understood why default weights in both functions are identical for charged hadrons (211), similar for photons (22) and quite different for neutral hadrons (130)?

Reply:
> BTW, is it understood why default weights in both functions are identical for charged hadrons (211), similar for photons (22) and quite different for neutral hadrons (130)?

@lwezenbe what are the origins of the default values for cases where information is missing? Did you take them from Maria Teresa's code? I remember that the values she used were preliminary and maybe for the central CMSSW integration we should re-evaluate them more accurately. Also, the mean values must be different for inner and outer cells (this was not implemented in Maria Teresa's version).

Reply:
Actually, I think I probably added them myself when changing some defaults. In Maria-Teresa's code, she had only AOD, so she could just set the value in the "get(...) = " lines themselves, as here for example:
https://github.com/mgrippo/cmssw/blob/deepTau_hlt_dev/RecoTauTag/RecoTau/plugins/DeepTauId.cc#L1319
So if we need to re-evaluate even more accurately, how should we go about it for now?

Reply:
Best way is to extract values directly from DeepTau v2 training root-tuples (currently being copied to /eos/cms/store/group/phys_tau/TauML/DeepTau_v2/tuples-v2-training-v2-t1-root/training/).
E.g.
```python
file = '/eos/cms/store/group/phys_tau/TauML/DeepTau_v2/tuples-v2-training-v2-t1-root/training/part_0.root'
tree = 'inner_cells'
val_flag = 'pfCand_ele_valid'
var = 'pfCand_ele_puppiWeight'

df = ROOT.RDataFrame(tree, file)
x = df.Filter('{} > 0'.format(val_flag)).Mean(var)
print('E[{}] = {}'.format(var, x.GetValue()))
```